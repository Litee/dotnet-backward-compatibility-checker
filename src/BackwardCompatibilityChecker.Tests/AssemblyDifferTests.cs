using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using BackwardCompatibilityChecker.Introspection.Diff;
using BackwardCompatibilityChecker.Introspection.Query;
using Mono.Cecil;
using NUnit.Framework;

namespace BackwardCompatibilityChecker.Tests
{
    [UseReporter(typeof(WinMergeReporter))]
    [TestFixture]
    public class AssemblyDifferTests
    {
        [Test]
        public void ShouldFindDifferences()
        {
            var assemblyV1 = AssemblyDefinition.ReadAssembly(@"..\..\..\OldAssembly\bin\Debug\OldAssembly.dll");
            var assemblyV2 = AssemblyDefinition.ReadAssembly(@"..\..\..\NewAssembly\bin\Debug\OldAssembly.dll");
            var assemblyDiffer = new AssemblyDiffer(assemblyV1, assemblyV2);
            var assemblyDiffCollection = assemblyDiffer.GenerateTypeDiff(QueryAggregator.PublicApiQueries);
            var stringWriter = new StringWriter();
            new DiffPrinter(stringWriter).Print(assemblyDiffCollection);
            Approvals.Verify(stringWriter.ToString());
        }
    }
}