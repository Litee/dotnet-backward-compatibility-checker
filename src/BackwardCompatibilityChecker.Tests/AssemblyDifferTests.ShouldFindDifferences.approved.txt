	public class Assembly.Class1
		- public event System.EventHandler EventThatWillBeRemoved
		+ public int get_PropertyThatWillChangeType()
		+ public string MethodThatWillChangeReturnType()
		+ public void MethodThatWillGetNewParameter(Int32 x,String y)
		- public string get_PropertyThatWillBeRemoved()
		- public string get_PropertyThatWillChangeType()
		- public void MethodThatWillBeRemoved()
		- public void MethodThatWillGetNewParameter(Int32 x)
		- public bool MethodThatWillChangeReturnType()
